libvast/src/detail/settings.cpp

```diff
@@ -13,33 +13,57 @@
 
 #include "vast/detail/settings.hpp"
 
+#include "vast/detail/type_traits.hpp"
 #include "vast/logger.hpp"
 
 namespace vast::detail {
 
 namespace {
 
+template <class Policy>
 void merge_settings_impl(const caf::settings& src, caf::settings& dst,
-                         size_t depth = 0) {
+                         Policy policy, size_t depth = 0) {
   if (depth > 100) {
     VAST_ERROR_ANON("Exceeded maximum nesting depth in settings.");
     return;
   }
   for (auto& [key, value] : src) {
-    if (caf::holds_alternative<caf::settings>(value))
+    if (caf::holds_alternative<caf::settings>(value)) {
       merge_settings_impl(caf::get<caf::settings>(value),
-                          dst[key].as_dictionary(), depth + 1);
-    else
-      dst.insert_or_assign(key, value);
+                          dst[key].as_dictionary(), policy, depth + 1);
+    } else {
+      if constexpr (std::is_same_v<Policy, policy::deep_tag>) {
+        if (caf::holds_alternative<caf::config_value::list>(value)) {
+          const auto& src_list = caf::get<caf::config_value::list>(value);
+          if (caf::holds_alternative<caf::config_value::list>(dst[key])) {
+            auto& dst_list = dst[key].as_list();
+            dst_list.insert(dst_list.end(), src_list.begin(), src_list.end());
+          } else {
+            dst.insert_or_assign(key, src_list);
+          }
+        } else {
+          dst.insert_or_assign(key, value);
+        }
```

Comment:
Not sure this will work correctly:
```suggestion
      if constexpr (std::is_same_v<Policy, policy::deep_tag>) {
        if (caf::holds_alternative<caf::config_value::list>(value) &&
            caf::holds_alternative<caf::config_value::list>(dst[key])) {
          const auto& src_list = caf::get<caf::config_value::list>(value);
          auto& dst_list = dst[key].as_list();
          dst_list.insert(dst_list.end(), src_list.begin(), src_list.end());
        } else {
          dst.insert_or_assign(key, value);
        }
```